compiler/rustc_target/src/spec/mipsel_unknown_none.rs

```diff
@@ -0,0 +1,32 @@
+// Bare MIPS32r2, little endian, softfloat
+// can be used for MIPS M4K core (e.g. on PIC32MX devices)
+
+use crate::spec::{LinkerFlavor, LldFlavor, RelocModel};
+use crate::spec::{PanicStrategy, Target, TargetOptions};
+
+pub fn target() -> Target {
+    Target {
+        llvm_target: "mipsel-unknown-none".to_string(),
+        target_endian: "little".to_string(),
+        pointer_width: 32,
+        target_c_int_width: "32".to_string(),
+        data_layout: "e-m:m-p:32:32-i8:8:32-i16:16:32-i64:64-n32-S64".to_string(),
+        arch: "mips".to_string(),
+        target_os: "none".to_string(),
+        target_env: String::new(),
+        target_vendor: String::new(),
+        linker_flavor: LinkerFlavor::Lld(LldFlavor::Ld),
+
+        options: TargetOptions {
+            cpu: "mips32r2".to_string(),
+            features: "+mips32r2,+soft-float,+noabicalls".to_string(),
+            max_atomic_width: Some(32),
+            executables: true,
+            linker: Some("rust-lld".to_owned()),
+            panic_strategy: PanicStrategy::Abort,
+            relocation_model: RelocModel::Static,
+            emit_debug_gdb_scripts: false,
+            ..Default::default()
+        },
```

Comment:
By default, targets allow all ABIs, but the thumb target blacklist the following:

    vec![Abi::Stdcall, Abi::Fastcall, Abi::Vectorcall, Abi::Thiscall, Abi::Win64, Abi::SysV64]

Which ones are supported by LLVM when targeting MIPS? We might also need such a blacklist here.

Reply:
Not absolutely sure about this. I understand that these are calling conventions that are not supported by MIPS (mainly x86 specific). But I wonder why they are not blacklisted in the other MIPS targets. From my point of view it would make sense to blacklist them. I tried a so modified rustc and it works for a PIC32MX.

I propose to add the blacklist. Do you agree?

Reply:
Yeah.

For the PSP target, I probably just forgot to check for that (since I also reviewed that target), so we might want to add the blacklist there as well.

Reply:
Done.